### DISCUSSION TITLE
About Self-Conditioning

### DISCUSSION THREAD

**Discussion by v-shaoningli** [RosettaCommons/RFdiffusion#98] [General]
Hi all! Thanks for your excellent work **RFdiffusion**. 

Here I have one question about the implementation of `Self-Conditioning`, i.e., Algorithm 3 in the supplementary.
Compare with the original paper _Analog bits: Generating discrete data using diffusion models with self-conditioning_,  RFdiffusion additionally added `ReverseStep` to generate $x^{t}$ from $x^{t+1}$.

To be concrete, I attach the figure in the following:
<img width="828" alt="image" src="https://github.com/RosettaCommons/RFdiffusion/assets/108473158/8c721c28-54e8-4f9b-9386-96fc98e92e49">

What RFdiffusion is done: 
<img width="503" alt="image" src="https://github.com/RosettaCommons/RFdiffusion/assets/108473158/86021bf1-a703-4a9a-b3e7-ad0fb12e5709">
1. Generate $x^{t+\Delta}$;
2. Employ `ReversteStep` to generate $x^{t}$;
3. Use RFdiffusion to predict $\tilde{x}_0$ and stop gradient;
4. Use $x^{t}$ and $\tilde{x}_0$ as input of RFdiffusion to predict $x^{pred}$.

Instead, in the original paper:
<img width="510" alt="image" src="https://github.com/RosettaCommons/RFdiffusion/assets/108473158/018079ea-0cd5-40cd-8b8e-becba0822013">
1. Generate $x^{t+\Delta}$;
2. Use NeuralNet to predict $\tilde{x}_0$ and stop gradient;
3. Use $x^{t+\Delta}$ and $\tilde{x}_0$ as input of NeuralNet to predict $x^{pred}$.

RFdiffusion has one additional step. Is it more benifitial to predict $x^{pred}$ with $x^{t}$ and $\tilde{x}_0$ compared with $x^{t+\Delta}$ and $\tilde{x}_0$ (because employing `ReversteStep` is a little bit more time-consuming)?

Looking forward to your responses. Thanks.

Best,
Shaoning

**Comment by KatarinaYuan** [comment]
Wondering the same thing here. Is this design better than the self-conditioning used in bit diffusion? How much is the performance gain?
